Java/ds/Binary_Tree.java

```diff
@@ -37,13 +37,37 @@ public Node takeInput(Node parent,Boolean isleftorright){
         if(choice){
             node.right=takeInput(node,false);
         }
+        System.out.println("Do you want tree traversal ");
+        boolean chose = s.nextBoolean();
+        if(chose){
+            traversal();
+        }
         return node;
     }
+    public void traversal()
+    {
+        System.out.println("Enter 1 for preorder traversal or enter 2 for postorder traversal ");
+        int option = s.nextInt();
+        switch(option)
+        {
+            case 1:
+            preOrder();
+            break;
+
+            case 2:
+            postOrder();
+            break;
+        }
+
+    }
 
     public void preOrder() {
```

Comment:
why calling a function and that calling another functinon

Reply:
i had done the requested changes please review it